src/functions/template-tags/general.php

```diff
@@ -704,3 +704,100 @@ function tribe_is_frontend() {
 		return (bool) apply_filters( 'tribe_doing_frontend', false );
 	}
 }
+
+if ( ! function_exists( 'tribe_post_checksum' ) ) {
+	/**
+	 * Builds a post checksum based on a list of fields that are
+	 * set to change when relevant post details change.
+	 *
+	 * This function will only calculate the checksum: it will not perform any
+	 * kind of check or monitoring on the post itself.
+	 *
+	 * @since TBD
+	 *
+	 * @param int|WP_Post $post_id   The post ID or object
+	 * @param array       $use_frags An array detailing the post fields, columns of
+	 *                               the `posts` table, that should be used to build
+	 *                               the checksum; defaults to the post `ID` and `post_modified`
+	 *                               columns.
+	 *
+	 * @return null|string The checksum string or `null` if the post is not valid or no
+	 *                     fragments are passed.
+	 */
+	function tribe_post_checksum( $post_id, $use_frags = array( 'ID', 'post_modified' ) ) {
+		if ( empty( $post_id ) || empty( $use_frags ) ) {
+			return null;
+		}
+
+		$post = get_post( $post_id );
```

Comment:
If $post_id is already an instance of WP_Post, is this function call necessary?

Reply:
It's just convenient; if it's already an instance of `WP_Post` it will be returned: I would simply be duplicating the check made in the `get_post` function doing it here.